docs: update for name and destination changes (#641)
timescale/pgai: docs/vectorizer/api-reference.md

```

This function call:
1. Sets up a vectorizer named 'website_blog_vectorizer' for the `website.blog` table.
2. Creates a separate table `website.blog_embeddings_store` to store embeddings.
3. Creates a view `website.blog_embeddings` joining the source and embeddings.
4. Loads the `contents` column.
5. Uses the Ollama `nomic-embed-text` model to create 768 dimensional embeddings.
6. Chunks the content into 128-character pieces with a 10-character overlap.
7. Formats each chunk with a `title` and a `published` date.
8. Grants necessary permissions to the roles `bob` and `alice`.

#### Example 2: Column destination

> For simpler cases where you need just one embedding per row (e.g. because your application already did the chunking), you can add a column directly to the source table:

Reviewer: ```suggestion
Column destination place the embedding in a separate column in the source table. It can only be used when the vectorizer does not perform chunking because it requires a one-to-one relationship between the source data and the embedding. This is useful in cases where you know the source text is short (as is common if the chunking has already been done upstream in your data pipeline).

The workflow is that your application inserts data into the table with a NULL in the embedding column. The vectorizer will then read the row, generate the embedding and update the row with the correct value in the embedding column.
```

```sql
SELECT ai.create_vectorizer(
    'website.product_descriptions'::regclass,
    name => 'product_descriptions_vectorizer',
    loading => ai.loading_column('description'),
    destination => ai.destination_column('description_embedding'),
    embedding => ai.embedding_openai('text-embedding-3-small', 768),
    chunking => ai.chunking_none(),  -- Required for column destination
    grant_to => ai.grant_to('marketing_team')
);
```

This function call:
1. Sets up a vectorizer named 'product_descriptions_vectorizer' for the `website.product_descriptions` table.
2. Adds a column called `description_embedding` directly to the source table.

> 3. Loads the `description` column.
> 4. Uses OpenAI's embedding model to create 768 dimensional embeddings.
> 5. Doesn't chunk the content (required for column destination).
> 6. Grants necessary permissions to the role `marketing_team`.

Reviewer: ```suggestion
3. Loads the `description` column.
4. Doesn't chunk the content (required for column destination).
5. Uses OpenAI's embedding model to create 768 dimensional embeddings.
6. Writes the embedding into the `description_embedding` column by updating the row.
7. Grants necessary permissions to the role `marketing_team`.
```

| Name             | Type                                                   | Default                           | Required | Description                                                                                        |
|------------------|--------------------------------------------------------|-----------------------------------|----------|----------------------------------------------------------------------------------------------------|
| source           | regclass                                               | -                                 | ✔        | The source table that embeddings are generated for.                                                |

> | name             | text                                                   | Auto-generated                    | ✖        | A unique name for the vectorizer. If not provided, it's auto-generated based on the destination type:<br>- For table destination: `[target_schema]_[target_table]`<br>- For column destination: `[source_schema]_[source_table]_[embedding_column]`<br>Must follow snake_case pattern `^[a-z][a-z_0-9]*$`                              |

Reviewer: don't we need to add the `if_not_exists` parameter here as well?

Author: yeh I forgot, added!

    ),
    -- other parameters...
);
```

For simpler configuration with defaults:

```sql
SELECT ai.create_vectorizer(
    'my_table'::regclass,
    destination => ai.destination_table('my_table_embeddings'),
    -- other parameters...
);
```

#### Parameters

`ai.destination_table` takes the following parameters:

| Name | Type | Default | Required | Description |
|------|------|---------|----------|-------------|
| destination | name | - | ✖ | The base name for the view and table. The view is named `<destination>`, the embedding table is named `<destination>_store`. |
| target_schema | name | Source table schema | ✖ | The schema where the embeddings table will be created. |
| target_table | name | `<source_table>_embedding_store` or `<destination>_store` | ✖ | The name of the table where embeddings will be stored. |
| view_schema | name | Source table schema | ✖ | The schema where the view will be created. |
| view_name | name | `<source_table>_embedding` or `<destination>` | ✖ | The name of the view that joins source and embeddings tables. |

#### Returns

A JSON configuration object that you can use in [ai.create_vectorizer](#create-vectorizers).

### ai.destination_column

> You use `ai.destination_column` to store embeddings directly in the source table as a new column. This approach:
> - Adds a vector column directly to the source table
> - Does not create a separate view
> - Requires chunking to be set to `ai.chunking_none()` (no chunking)
> - Stores exactly one embedding per row

Reviewer: ```suggestion
You use `ai.destination_column` to store embeddings directly in the source table as a new column. This approach can only be used when the vectorizer does not perform chunking because it requires a one-to-one relationship between the source data and the embedding. This is useful in cases where you know the source text is short (as is common if the chunking has already been done upstream in your data pipeline).

This approach:
- Adds a vector column directly to the source table
- Does not create a separate view
- Requires chunking to be set to `ai.chunking_none()` (no chunking)
- Stores exactly one embedding per row

The workflow is that your application inserts data into the table with a NULL in the embedding column. The vectorizer will then read the row, generate the embedding and update the row with the correct value in the embedding column.
```

#### Example usage

To resume the automatic scheduling for a vectorizer:

```sql
-- Using ID
SELECT ai.enable_vectorizer_schedule(1);

-- Using name (recommended)
SELECT ai.enable_vectorizer_schedule('public_blog_embeddings');
```

#### Parameters

> `ai.enable_vectorizer_schedule` takes one of the following parameters:

Reviewer: misleading... there are two different functions

#### Example usage

To stop the automatic scheduling for a vectorizer:

```sql
-- Using ID
SELECT ai.disable_vectorizer_schedule(1);

-- Using name (recommended)
SELECT ai.disable_vectorizer_schedule('public_blog_embeddings');
```

#### Parameters

> `ai.disable_vectorizer_schedule` takes one of the following parameters:

Reviewer: this is misleading. There are now two functions one that takes an id and one that takes a name. Note one that takes an id or a name.

Author: okay I felt like two tables is more confusing than this, but changed it everywhere, let me know what you think.

|Name| Type | Default | Required | Description |
|-|------|-|-|-|

> |vectorizer_id| int  | -|✔*|The identifier of the vectorizer you want to drop|

Reviewer: again having both in the same table is misleading

To get an exact count, regardless of queue size, set the optional parameter to
`true` like this:

```sql
-- Using ID
SELECT ai.vectorizer_queue_pending(1, exact_count=>true);

-- Using name (recommended)
SELECT ai.vectorizer_queue_pending('public_blog_embeddings', exact_count=>true);
```

#### Parameters

`ai.vectorizer_queue_pending function` takes the following parameters:

| Name          | Type | Default | Required | Description                                             |
|---------------|------|---------|----------|---------------------------------------------------------|

> | vectorizer_id | int  | -       | ✔*       | The identifier of the vectorizer you want to check      |

Reviewer: misleading
